packages/client/components/EditorLinkChanger/EditorLinkChangerDraftjs.tsx

```diff
@@ -37,11 +51,49 @@ const EditorLinkChangerDraftjs = (props: Props) => {
   } = props
   useTaskChild('editor-link-changer')
   const handleSubmit = ({text, href}: {text: string; href: string}) => {
+    const keepSelection = true
+    // don't include leading and trailing whitespace in the link text
+    const nonWhitespaceFromStart = text.search(/\S/)
+    const nonWhitespaceFromEnd = text.search(/\s*$/)
+    const startStr = nonWhitespaceFromStart === -1 ? '' : text.slice(0, nonWhitespaceFromStart)
+    const endStr = nonWhitespaceFromEnd === -1 ? '' : text.slice(nonWhitespaceFromEnd)
+    const trimmedText = text.trim() ? text.trim() : text
     const focusedEditorState = EditorState.forceSelection(editorState, selectionState)
-    const nextEditorState = completeEntity(focusedEditorState, 'LINK', {href}, text, {
-      keepSelection: true
+    let newEditorState = focusedEditorState
+    // For whitespace only links
+    if (text.trim()) {
```

Comment:
-1 This is the non-whitespace branch, so the comment above is a bit confusing. Since we branch more often, maybe it's helpful to put this in an variable to make it explicit, so we can avoid some of the comments. This should then also be used in l60
```ts
const hasTextTitle = !!text.trim()
```

Reply:
I am not getting what you mean by branch here (maybe it is a naive question😑). Could you define this term?

Reply:
You're branching the execution flow based on a condition
```
if (condition) {
  // "true branch"
} else {
  // "false branch"
}
```
So anytime something happens on a condition, there are multiple execution branches.

What I meant is, because the `text.trim()` appeared in 3 different conditions (`if` and `?`) let's just put it in a variable with a meaningful name so we can omit comments.

Reply:
Using
```
hasTextTitle = text.trim()
```